### ISSUE TITLE
Test failures with cairo 1.17.6

### ISSUE THREAD

**Issue by felixonmars** [Kozea/cairocffi#204]
All seem to be PDF related:

```
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.0.0
rootdir: /build/python-cairocffi/src/cairocffi-1.3.0, configfile: setup.cfg
collected 54 items

build/lib/cairocffi/test_cairo.py ...........FFFF.F.FF.................. [ 70%]
........                                                                 [ 85%]
build/lib/cairocffi/test_numpy.py .                                      [ 87%]
build/lib/cairocffi/test_pixbuf.py .x...                                 [ 96%]
build/lib/cairocffi/test_xcb.py ss                                       [100%]

=================================== FAILURES ===================================
________________________________ test_metadata _________________________________

    @pytest.mark.xfail(cairo_version() < 11504,
                       reason='Cairo version too low')
    def test_metadata():
        file_obj = io.BytesIO()
        surface = PDFSurface(file_obj, 1, 1)
        surface.set_metadata(PDF_METADATA_TITLE, 'title')
        surface.set_metadata(PDF_METADATA_SUBJECT, 'subject')
        surface.set_metadata(PDF_METADATA_CREATOR, 'creator')
        surface.set_metadata(PDF_METADATA_AUTHOR, 'author')
        surface.set_metadata(PDF_METADATA_KEYWORDS, 'keywords')
        surface.set_metadata(PDF_METADATA_CREATE_DATE, '2013-07-21T23:46:00+01:00')
        surface.set_metadata(PDF_METADATA_MOD_DATE, '2013-07-21T23:46:00Z')
        surface.finish()
        pdf_bytes = file_obj.getvalue()
>       assert b'/Title (title)' in pdf_bytes
E       AssertionError: assert b'/Title (title)' in b'%PDF-1.7\n%\xb5\xed\xae\xfb\n4 0 obj\n<< /Length 5 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C a\xa8\x...x82S\xc0\x98\xa2\x89!\xf6\xc7\x968Z\x12\x13nQ\x7f\x8b\x0e\x0f\xe5\xd3\x05b\nendstream\nendobj\nstartxref\n801\n%%EOF\n'

build/lib/cairocffi/test_cairo.py:253: AssertionError
_________________________________ test_outline _________________________________

    @pytest.mark.xfail(cairo_version() < 11504,
                       reason='Cairo version too low')
    def test_outline():
        file_obj = io.BytesIO()
        surface = PDFSurface(file_obj, 1, 1)
        outline = surface.add_outline(
            PDF_OUTLINE_ROOT, 'title 1', 'page=1 pos=[1 1]',
            PDF_OUTLINE_FLAG_OPEN & PDF_OUTLINE_FLAG_BOLD)
        surface.add_outline(outline, 'title 2', 'page=1 pos=[1 1]')
        surface.finish()
        pdf_bytes = file_obj.getvalue()
>       assert b'/Title (title 1)' in pdf_bytes
E       AssertionError: assert b'/Title (title 1)' in b'%PDF-1.7\n%\xb5\xed\xae\xfb\n6 0 obj\n<< /Length 7 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C a\xa8\x...B\x143\x82\x08\x16\x10\xc1\xca\xc8l\x08\x94`\xf6``\x00\x00\x1f\xc7\x05\xea\nendstream\nendobj\nstartxref\n840\n%%EOF\n'

build/lib/cairocffi/test_cairo.py:275: AssertionError
_______________________________ test_page_label ________________________________

    @pytest.mark.xfail(cairo_version() < 11504,
                       reason='Cairo version too low')
    def test_page_label():
        file_obj = io.BytesIO()
        surface = PDFSurface(file_obj, 1, 1)
        surface.set_page_label('abc')
        surface.finish()
        pdf_bytes = file_obj.getvalue()
>       assert b'/P (abc)' in pdf_bytes
E       AssertionError: assert b'/P (abc)' in b'%PDF-1.7\n%\xb5\xed\xae\xfb\n4 0 obj\n<< /Length 5 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C a\xa8\x...e040\xa6\xe8b\x88]\xd8\x12G\x93\xc4\x84\x8b0\xbf\x15\xa2\xbe\x1f\x07X\x06<\nendstream\nendobj\nstartxref\n776\n%%EOF\n'

build/lib/cairocffi/test_cairo.py:287: AssertionError
___________________________________ test_tag ___________________________________

    @pytest.mark.xfail(cairo_version() < 11504,
                       reason='Cairo version too low')
    def test_tag():
        file_obj = io.BytesIO()
        surface = PDFSurface(file_obj, 10, 10)
        context = Context(surface)
        context.tag_begin('Document')
        context.tag_begin(
            TAG_LINK,
            attributes='rect=[1 2 4 5] uri=\'https://cairocffi.readthedocs.io/\'')
        context.set_source_rgba(1, 0, .5, 1)
        context.rectangle(2, 3, 4, 5)
        context.fill()
        context.tag_end(TAG_LINK)
        context.tag_end('Document')
        context.show_page()
        surface.finish()
        pdf_bytes = file_obj.getvalue()
>       assert b'/URI (https://cairocffi.readthedocs.io/)' in pdf_bytes
E       assert b'/URI (https://cairocffi.readthedocs.io/)' in b'%PDF-1.7\n%\xb5\xed\xae\xfb\n6 0 obj\n<< /Length 7 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C ah\xa0\...\xdd\xdc\x85\xb6\xffe \xc5U]<>[\x85"\xcdl\x16\xe5Qk\x9e\xf0\x01i\x99\x06t\nendstream\nendobj\nstartxref\n1151\n%%EOF\n'

build/lib/cairocffi/test_cairo.py:308: AssertionError
______________________________ test_document_unit ______________________________

    @pytest.mark.xfail(cairo_version() < 11510,
                       reason='Cairo version too low')
    def test_document_unit():
        surface = SVGSurface(None, 1, 2)
>       assert surface.get_document_unit() == SVG_UNIT_PT
E       assert 0 == 7
E        +  where 0 = <bound method SVGSurface.get_document_unit of <cairocffi.surfaces.SVGSurface object at 0x7f6028d6bdf0>>()
E        +    where <bound method SVGSurface.get_document_unit of <cairocffi.surfaces.SVGSurface object at 0x7f6028d6bdf0>> = <cairocffi.surfaces.SVGSurface object at 0x7f6028d6bdf0>.get_document_unit

build/lib/cairocffi/test_cairo.py:334: AssertionError
______________________________ test_pdf_versions _______________________________

    @pytest.mark.xfail(cairo_version() < 11000,
                       reason='Cairo version too low')
    def test_pdf_versions():
        assert set(PDFSurface.get_versions()) >= set([
            cairocffi.PDF_VERSION_1_4, cairocffi.PDF_VERSION_1_5])
        assert PDFSurface.version_to_string(cairocffi.PDF_VERSION_1_4) == 'PDF 1.4'
        with pytest.raises(TypeError):
            PDFSurface.version_to_string('PDF_VERSION_42')
        with pytest.raises(ValueError):
            PDFSurface.version_to_string(42)

        file_obj = io.BytesIO()
        PDFSurface(file_obj, 1, 1).finish()
>       assert file_obj.getvalue().startswith(b'%PDF-1.5')
E       AssertionError: assert False
E        +  where False = <built-in method startswith of bytes object at 0x557026e2ec70>(b'%PDF-1.5')
E        +    where <built-in method startswith of bytes object at 0x557026e2ec70> = b'%PDF-1.7\n%\xb5\xed\xae\xfb\n4 0 obj\n<< /Length 5 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C a\xa8\x...\xa2@\xd0M5\xcd\xccO\x0c\xb3\xbc$a\x14\xda\xbf\xe9\xc0\x03\xee\x11\x06\xe2\nendstream\nendobj\nstartxref\n738\n%%EOF\n'.startswith
E        +      where b'%PDF-1.7\n%\xb5\xed\xae\xfb\n4 0 obj\n<< /Length 5 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C a\xa8\x...\xa2@\xd0M5\xcd\xccO\x0c\xb3\xbc$a\x14\xda\xbf\xe9\xc0\x03\xee\x11\x06\xe2\nendstream\nendobj\nstartxref\n738\n%%EOF\n' = <built-in method getvalue of _io.BytesIO object at 0x7f6028be3740>()
E        +        where <built-in method getvalue of _io.BytesIO object at 0x7f6028be3740> = <_io.BytesIO object at 0x7f6028be3740>.getvalue

build/lib/cairocffi/test_cairo.py:411: AssertionError
_______________________________ test_pdf_surface _______________________________

    def test_pdf_surface():
        with temp_directory() as tempdir:
            filename = os.path.join(tempdir, 'foo.pdf')
            filename_bytes = filename.encode(sys.getfilesystemencoding())
            file_obj = io.BytesIO()
            for target in [filename, filename_bytes, file_obj, None]:
                surface = PDFSurface(target, 123, 432)
                surface.finish()
            with open(filename, 'rb') as fd:
                assert fd.read().startswith(b'%PDF')
            with open(filename_bytes, 'rb') as fd:
                assert fd.read().startswith(b'%PDF')
            pdf_bytes = file_obj.getvalue()
            assert pdf_bytes.startswith(b'%PDF')
>           assert b'/MediaBox [ 0 0 123 432 ]' in pdf_bytes
E           assert b'/MediaBox [ 0 0 123 432 ]' in b"%PDF-1.7\n%\xb5\xed\xae\xfb\n4 0 obj\n<< /Length 5 0 R\n   /Filter /FlateDecode\n>>\nstream\nx\x9c3T0\x00B]C abl\xa4...x11\xbbAJ8\x18\x18A\x04\x13#\xd3%\xa0\x18\xd3K\x06\x06\x00\xf0\xb1\x07\x04\nendstream\nendobj\nstartxref\n745\n%%EOF\n"

build/lib/cairocffi/test_cairo.py:434: AssertionError
=========================== short test summary info ============================
FAILED build/lib/cairocffi/test_cairo.py::test_metadata - AssertionError: ass...
FAILED build/lib/cairocffi/test_cairo.py::test_outline - AssertionError: asse...
FAILED build/lib/cairocffi/test_cairo.py::test_page_label - AssertionError: a...
FAILED build/lib/cairocffi/test_cairo.py::test_tag - assert b'/URI (https://c...
FAILED build/lib/cairocffi/test_cairo.py::test_document_unit - assert 0 == 7
FAILED build/lib/cairocffi/test_cairo.py::test_pdf_versions - AssertionError:...
FAILED build/lib/cairocffi/test_cairo.py::test_pdf_surface - assert b'/MediaB...
============== 7 failed, 44 passed, 2 skipped, 1 xfailed in 0.39s ==============
```

**Comment by liZe** [comment]
It looks like the metadata are now stored as a compressed string. As far as I can tell, there’s now no easy way to test that they are in the PDF. If anyone has an idea…

**Comment by felixonmars** [comment]
I'm not sure, but would you consider using another python pdf library to inspect the data? (Of course it's only used in the tests).

**Comment by liZe** [comment]
> I'm not sure, but would you consider using another python pdf library to inspect the data? (Of course it's only used in the tests).

Well, why not!

**Comment by liZe** [comment]
Fixed by #205.
